roles/check-swlib/tasks/main.yml

```diff
@@ -13,12 +13,17 @@
 # limitations under the License.
 
 ---
-- name: check-swlib | Look up Oracle release
-  #if release set to latest then get most recent release from rdbms_patches array
-  #if nothing returned then set to "base"
+- name: check-swlib | Define Oracle release
+  #if release set to latest then get most recent release from gi_patches array
+  #if nothing returned then set to "base" in next task
   set_fact:
-    oracle_rel: "{% if oracle_rel == 'latest' %}{{ gi_patches | json_query('[?base==`' + oracle_ver + '`].release')  | sort(reverse=true) | first | default('base', true) }}{% elif oracle_rel == 'base' %}{% else %}{{ oracle_rel }}{% endif %}"
-  when: oracle_rel is defined
+    oracle_rel: "{% if not ansible_loop.first %}{% if item is version(oracle_rel, '>=') %}{{ item }}{% else %}{{ oracle_rel }}{% endif %}{% else %}{{ item }}{% endif %}"
+  loop: "{{ gi_patches | selectattr('base', '==', oracle_ver) | map(attribute='release') | list }}"
```

Comment:
I used and successfully tested the following:
    # Using expression from bms-toolkit/roles/rac-db-setup/tasks/rac-db-install.yml Line 46
    highest_version_num: "{% if not ansible_loop.first %}{% if item is version(highest_version_num, '>=') %}{{ item }}{% else %}{{ highest_version_num }}{% endif %}{% else %}{{ item }}{% endif %}"
  loop: "{{ gi_patches | json_query('[?base==`' + oracle_ver + '`].release') }}"

Not sure on the array output by the following loop:
```
  loop: "{{ gi_patches | selectattr('base', '==', oracle_ver) | map(attribute='release') | list }}"
```

Instead of introducing a new one, will it be ok to see if we can keep the same `loop` for consistency (especially given we are using the same jinja expression) and reuse the one from: 
https://github.com/google/bms-toolkit/blob/master/roles/rac-db-setup/tasks/rac-db-install.yml#L51

Reply:
Ahh, yes, I have a bigger change I'm working on to consolidate into a single definition of oracle_rel (as we currently have two, and one is actually wrong).  But I agree that introducing a third doesn't look right from a review perspective.  I'll port the larger changes into this branch so that it covers all cases of oracle_rel checks.

As to the logic, I'll put together some testcases to show that this logic is correct, especially across versions with multiple digits.

Reply:
Update:  in order to avoid testing the full patch workflow, for now I'm going to try and keep the scope small and jsut include the existing patch/ logic.  I'll address the overall oracle_rel logic in a future PR.